action.yml

```diff
@@ -17,66 +16,61 @@ inputs:
     description: --resource option
     required: true
     default: ""
-  image-tag:
-    description: Image tag to build and push (default is github.sha)
-    required: true
-    default: ""
 outputs: {}
 runs:
   using: composite
   steps:
-    - uses: freckle/setup-platform-action@v4
+    - uses: freckle/setup-platform-action@v6
       with:
-        version: ${{ inputs.platform-version }}
-        suffix: ${{ inputs.platform-suffix }}
-
-    - shell: bash
-      run: platform container:login
+        version: ${{ inputs.platform-setup-version }}
+        token: ${{ inputs.platform-setup-token }}
 
     - id: prep
       shell: bash
       run: |
         cd '${{ inputs.app-directory }}'
 
-        if [[ -n '${{ inputs.app-resource }}' ]]; then
-          query_setting() {
-            platform --resource '${{ inputs.app-resource }}' query "settings.$1"
-          }
-        else
-          query_setting() {
-            platform query "settings.$1"
-          }
-        fi
+        # Set each setting as an output of the same name
+        for name in Dockerfile DockerBuildContext EcrRepository; do
+          printf "::set-output name=$name::"
 
-        context=${{ inputs.app-directory }}/$(query_setting DockerBuildContext)
-        echo "::set-output name=context::$context"
+          if [[ -n '${{ inputs.app-resource }}' ]]; then
+            platform --resource '${{ inputs.app-resource }}' query "settings.$name"
+          else
+            platform query "settings.$name"
+          fi
+        done
 
-        dockerfile=$(query_setting Dockerfile)
-        echo "::set-output name=dockerfile::$dockerfile"
-
-        tag=${{ inputs.image-tag }}
-        tag=${tag:-${{ github.sha }}}
-        tags=$(query_setting EcrRepository):$tag
-        echo "::set-output name=tags::$tags"
+    - id: meta
+      uses: docker/metadata-action@v3
+      with:
+        images: ${{ steps.prep.outputs.EcrRepository }}
+        tags: |
+          type=sha,prefix=,format=long,priority=1000
```

Comment:
This is equivalent to `${{ github.sha }}`, which we have been using and will now only use (I removed the configurability). The reason I'm bringing in `docker/metadata-action` is that it will also handle labels and `:edge`.